packages/clay-checkbox/src/ClayCheckbox.tsx

```diff
@@ -0,0 +1,67 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import * as React from 'react';
+import classNames from 'classnames';
+
+interface Props extends React.InputHTMLAttributes<HTMLInputElement> {
+	containerProps?: React.HTMLAttributes<HTMLDivElement>;
+	indeterminate?: boolean;
+	inline?: boolean;
+	label?: React.ReactElement | React.ReactText;
+}
+
+const ClayCheckbox: React.FunctionComponent<Props> = ({
+	checked = false,
+	className,
+	containerProps = {className: ''},
+	indeterminate = false,
+	inline,
+	label,
+	...otherProps
+}) => {
+	const inputRef = React.useRef<HTMLInputElement | null>(null);
+
+	React.useEffect(() => {
+		if (inputRef.current) {
+			inputRef.current.indeterminate = indeterminate;
+		}
+	}, [indeterminate]);
+
+	return (
+		<div
+			{...containerProps}
+			className={classNames(
+				'custom-control custom-checkbox',
+				containerProps.className,
+				{
+					'custom-control-inline': inline,
+				}
+			)}
+		>
+			<label>
+				<input
+					{...otherProps}
+					checked={checked}
+					className={classNames('custom-control-input', className)}
+					ref={inputRef}
+					type="checkbox"
+					value={checked.toString()}
+				/>
+
+				{label && (
+					<span className="custom-control-label">
+						<span className="custom-control-label-text">
```

Comment:
At some point we will need to add the `sr-only` class. Previously we had `showLabel` API for this.

Reply:
I talked to @pat270 about this and we settled on removing `sr-only` and having people rely on using aria attributes.

Reply:
Hmm, we have to be careful about that. We have to add this information in the migration guide and also add as a good practice in the new documentation.